### PR TITLE
Implement temp directory

### PR DESCRIPTION
**Pull request by janhohenheim**

### CODE REVIEW

File: TempDirectory.Test/TempDirectoryBuilderTest.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                Assert.NotEqual(tempDirectory.FullName, tempDirectory.Name);
+            }
+        }
+
+        [Fact]
+        public void PrefixIsInName()
+        {
+            using (var tempDirectory = new TempDirectoryBuilder().Prefix(Prefix).Create())
+            {
+                Assert.Contains(Prefix, tempDirectory.Name);
```
Comment (tautropfli) [inline TempDirectory.Test/TempDirectoryBuilderTest.cs]: Test that the name starts with the prefix

File: TempDirectory.Test/TempDirectoryBuilderTest.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            }
+        }
+
+        [Fact]
+        public void SuffixIsInName()
+        {
+            using (var tempDirectory = new TempDirectoryBuilder().Suffix(Suffix).Create())
+            {
+                Assert.Contains(Suffix, tempDirectory.Name);
```
Comment (tautropfli) [inline TempDirectory.Test/TempDirectoryBuilderTest.cs]: Test that the name ends with the prefix

File: TempDirectory.Test/TempDirectoryBuilderTest.cs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        public void SeparatorsAreInName()
+        {
+            using (var tempDirectory = new TempDirectoryBuilder()
+                .Prefix(Prefix)
+                .PrefixSeparator(PrefixSeparator)
+                .Suffix(Prefix)
+                .SuffixSeparator(SuffixSeparator)
+                .Create())
+            {
+                Assert.Contains(PrefixSeparator, tempDirectory.Name);
```
Comment (tautropfli) [inline TempDirectory.Test/TempDirectoryBuilderTest.cs:80]: Same as the two comments above.

**Review by tautropfli — DISMISSED** [review]

File: TempDirectory.Test/TempDirectoryBuilderTest.cs (hunk 3 above)
Comment (janhohenheim) [reply TempDirectory.Test/TempDirectoryBuilderTest.cs:80]: Are you sure? I don't think the separators should be at the start or end of the string.

File: TempDirectory.Test/TempDirectoryBuilderTest.cs (hunk 2 above)
Comment (janhohenheim) [reply TempDirectory.Test/TempDirectoryBuilderTest.cs]: Addressed in https://github.com/messerli-informatik-ag/temp-directory/pull/1/commits/6793941234fe7ebc0904638a7c90d0a31a54c7be

File: TempDirectory.Test/TempDirectoryBuilderTest.cs (hunk 1 above)
Comment (janhohenheim) [reply TempDirectory.Test/TempDirectoryBuilderTest.cs]: Addressed in https://github.com/messerli-informatik-ag/temp-directory/pull/1/commits/6793941234fe7ebc0904638a7c90d0a31a54c7be

File: TempDirectory/TempDirectoryBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+        private string GenerateDirectoryName()
+        {
+            var guid = Guid.NewGuid().ToString();
+            var prefixSubstring = string.IsNullOrEmpty(_prefix) ? "" : _prefix + _prefixSeparator;
+            var suffixSubstring = string.IsNullOrEmpty(_suffix) ? "" : _suffix + _suffixSeparator;
+
+           return $"{prefixSubstring}{guid}{suffixSubstring}";
+        }
+        private static OnDispose CreateDirectoryDeletionFn(string directoryPath)
```
Comment (dngroth) [inline TempDirectory/TempDirectoryBuilder.cs]: Avoid Abbrevation
```suggestion
        private static OnDispose CreateDirectoryDeletionFunction(string directoryPath)
```

**Review by dngroth — CHANGES_REQUESTED** [review]

File: TempDirectory/TempDirectoryBuilder.cs (hunk 4 above)
Comment (janhohenheim) [reply TempDirectory/TempDirectoryBuilder.cs]: Addressed in https://github.com/messerli-informatik-ag/temp-directory/pull/1/commits/e33a40f121b184e6fe88126aac0e71e3d33d4c08

**Review by dngroth — APPROVED** [review]

**Pull request merged**